lib/src/sv_sign.c

```diff
@@ -878,16 +918,18 @@ signed_video_set_private_key(signed_video_t *self, const char *private_key, size
     // Temporally turn the PEM |private_key| into an EVP_PKEY and allocate memory for signatures.
     SV_THROW(openssl_private_key_malloc(self->sign_data, private_key, private_key_size));
     SV_THROW(openssl_read_pubkey_from_private_key(self->sign_data, &self->pem_public_key));
-
-    self->plugin_handle = sv_signing_plugin_session_setup(private_key, private_key_size);
-    SV_THROW_IF(!self->plugin_handle, SV_EXTERNAL_ERROR);
+    self->private_key = private_key;
+
+    // Try to initialize ONVIF; if it is not available or cannot be initialized,
+    // fallback to Signed Video signing plugin.
```

Comment:
The comment about "fallback" should be inside `initialize_onvif` since you do not take any actions regarding that out here.